src/browser/fake_network.js

```diff
@@ -18,23 +17,65 @@ const TWO_TO_32 = Math.pow(2, 32);
 const DHCP_MAGIC_COOKIE = 0x63825363;
 const V86_ASCII = [118, 56, 54];
 
+/* TCP 4-way close handshake
+ *
+ *          Initiator    Receiver
+ *     (active close)    (passive close)
+ *              :           :
+ *  ESTABLISHED |           | ESTABLISHED
+ *              |           |
+ *   FIN_WAIT_1 |--- FIN -->| CLOSE_WAIT    [1]
+ *              |           |
+ *   FIN_WAIT_2 |<-- ACK ---|               [2]
+ *              |           |
+ *    TIME_WAIT |<-- FIN ---| LAST_ACK      [3]
+ *              |           |
+ *              |--- ACK -->| CLOSED        [4]
+ *       CLOSED |           |
+ *
+ * - Receiver MAY combine [2] and [3] into a single ACK+FIN packet.
+ * - State CLOSING (simultaneous close from both ends) not implemented.
+ * - In [3], instead of TIME_WAIT go diretly to CLOSED after sending ACK.
+ * - TODO: what about state LISTEN? Neither WISP nor FETCH seem to support it.
+ */
 const TCP_STATE_CLOSED = "closed";
-const TCP_STATE_LISTEN = "listen";
+const TCP_STATE_SYN_RECEIVED = "syn-received";
+const TCP_STATE_SYN_SENT = "syn-sent";
+//const TCP_STATE_LISTEN = "listen";
 const TCP_STATE_ESTABLISHED = "established";
 const TCP_STATE_FIN_WAIT_1 = "fin-wait-1";
 const TCP_STATE_CLOSE_WAIT = "close-wait";
 const TCP_STATE_FIN_WAIT_2 = "fin-wait-2";
 const TCP_STATE_LAST_ACK = "last-ack";
-const TCP_STATE_CLOSING = "closing";
-const TCP_STATE_TIME_WAIT = "time-wait";
-const TCP_STATE_SYN_RECEIVED = "syn-received";
-const TCP_STATE_SYN_SENT = "syn-sent";
+//const TCP_STATE_CLOSING = "closing";
+//const TCP_STATE_TIME_WAIT = "time-wait";
+
+// source: RFC6335, 6. Port Number Ranges
+const TCP_DYNAMIC_PORT_START = 49152;
+const TCP_DYNAMIC_PORT_END   = 65535;
+const TCP_DYNAMIC_PORT_RANGE = TCP_DYNAMIC_PORT_END - TCP_DYNAMIC_PORT_START;
+
+const ETH_HEADER_SIZE     = 14;
+const ETH_PAYLOAD_OFFSET  = ETH_HEADER_SIZE;
+const ETH_PAYLOAD_SIZE    = 1500;
+const ETH_TRAILER_SIZE    = 4;
+const ETH_FRAME_SIZE      = ETH_HEADER_SIZE + ETH_PAYLOAD_SIZE + ETH_TRAILER_SIZE;
+const IPV4_HEADER_SIZE    = 20;
+const IPV4_PAYLOAD_OFFSET = ETH_PAYLOAD_OFFSET + IPV4_HEADER_SIZE;
+const IPV4_PAYLOAD_SIZE   = ETH_PAYLOAD_SIZE - IPV4_HEADER_SIZE;
+const UDP_HEADER_SIZE     = 8;
+const UDP_PAYLOAD_OFFSET  = IPV4_PAYLOAD_OFFSET + UDP_HEADER_SIZE;
+const UDP_PAYLOAD_SIZE    = IPV4_PAYLOAD_SIZE - UDP_HEADER_SIZE;
+const TCP_HEADER_SIZE     = 20;
+const TCP_PAYLOAD_OFFSET  = IPV4_PAYLOAD_OFFSET + TCP_HEADER_SIZE;
+const TCP_PAYLOAD_SIZE    = IPV4_PAYLOAD_SIZE - TCP_HEADER_SIZE;
+const ICMP_HEADER_SIZE    = 4;
 
 function a2ethaddr(bytes) {
     return [0,1,2,3,4,5].map((i) => bytes[i].toString(16)).map(x => x.length === 1 ? "0" + x : x).join(":");
 }
 
-function siptolong(s) {
+function siptolong(s) { // TODO: this function is not used anywhere, can it be removed?
```

Comment:
Nice spot, let's remove it.

Reply:
Done.

There's another one: TCPConnection.connect(), see my comment there.

It was orphaned when we recently removed fake_tcp_connect() and FetchNetworkAdapter.tcp_connect() in [this commit](https://github.com/copy/v86/pull/1178/commits/f8af3c6c868557ac9b6a91847bf9070fbb9a75b3).

Just so I understand correctly, all these funcions are related to public server sockets (opened in the guest) which are not supported by either wisp- or fetch-based networking?